src/app/views/datasets-new/create/CreateCantabularDatasetController.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { Component } from "react";
+import { push } from "react-router-redux";
+import PropTypes from "prop-types";
+
+import datasets from "../../../utilities/api-clients/datasets";
+import notifications from "../../../utilities/notifications";
+import url from "../../../utilities/url";
+import log from "../../../utilities/logging/log";
+
+const propTypes = {
+    dispatch: PropTypes.func.isRequired,
+    params: PropTypes.shape({
+        datasetID: PropTypes.string.isRequired,
+        format: PropTypes.string.isRequired
+    }),
+    location: PropTypes.shape({
+        pathname: PropTypes.string.isRequired
+    })
+};
+
+export class CreateCantabularDatasetController extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isPosting: false,
+            datasetID: "",
+            format: ""
+        };
+    }
+
+    componentWillMount = () => {
+        this.setStateFromParameters();
+    };
+
+    setStateFromParameters = () => {
+        this.setState({
+            datasetID: this.props.params.datasetID,
+            format: this.props.params.format
+        });
+    };
+
+    makeCreateDatasetPostBody = () => {
+        return {
+            type: this.state.format
+        };
+    };
+
+    handleCreateClick = event => {
+        event.preventDefault();
+        const datasetID = this.state.datasetID;
+        const postBody = this.makeCreateDatasetPostBody();
```

Comment:
I think function `makeCreateDatasetPostBody` is redundant here, we can access the state directly and create object 
```suggestion
        const postBody = { type: this.state.format };
```

Reply:
Agreed but I am using the function in a unit test to ensure the post body is correctly formed